### PR TITLE
Add Chain Indicator in Header

### PR DESCRIPTION
**Pull request by madhurg-epns**
Design - https://www.figma.com/file/PzAmXUMOmwS2ljxufwv44m/PUSH-Prototypes?node-id=5249%3A8936&t=UOoVnQI8TRebFm2g-1

### CODE REVIEW

**Comment by madhurg-epns** [comment]
@0xNilesh @mishramonalisha76 The design team has given a go ahead to the styling of the Chain Indicator, we can go for the PR review.

**Comment by mishramonalisha76** [comment]
@madhurg-epns  

 I guess in tablet view the design needs to be like the mobile version.
image

the dropdown get above the select option's bottom border. In the design in just below the bottom border line.
image
image

The icons of the ethereum and polygon chain seems different in figma.

The font weight and font color seems different for the selected option in figma
image


The padding between the icon and the arrow seems lesser in figma and make the border radius of the entire box 19px
image

File: src/components/ChainIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  const fetchCurrentChainId = async () => {
+    const currentChainId = await window.ethereum.request({
+      method: 'eth_chainId',
+    });
+    setCurrentChainId(currentChainId);
+  };
+
+  // handles network change request
+  async function handleChangeNetwork(chainId) {
+    const chainIds = appConfig.allowedNetworks;
```
Comment (mishramonalisha76) [inline src/components/ChainIndicator.tsx:40]: use optional chaining, to avoid undefined cases

File: src/components/ChainIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+            </DropdownItem>
+          )}
+        </Container>
+      )}
+    </>
+  );
+};
+
+// css styles
+const Container = styled.button`
```
Comment (mishramonalisha76) [inline src/components/ChainIndicator.tsx:165]: try to use the button from  sharedStyling v2 file

File: src/components/ChainIndicator.tsx [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  flex: 1,
+  flex-direction: row;
+  align-items: center;
+  display: flex;
+  @media (max-width: 992px) {
+    width: 100%;
+    margin-right: 20px;
+  }
+`;
+const CurrentChain = styled.span`
```
Comment (mishramonalisha76) [inline src/components/ChainIndicator.tsx:182]: try to use the span from sharedStyling v2 file

File: src/components/Dropdown.tsx [hunk 4]
```diff
@@ -107,52 +117,55 @@ function Dropdown(props) {
...
 // css styles
-const Container = styled.div`
-  display: flex;
-  flex-direction: column;
-  padding: 1rem;
-`;
-const DropdownItem = styled.div`
-  display: flex;
-  margin: 1rem;
+const SpanAddress = styled(Span)`
```
Comment (mishramonalisha76) [inline src/components/Dropdown.tsx:138]: try to use sharedStyling

**Review by mishramonalisha76 — CHANGES_REQUESTED** [review]

File: src/components/ChainIndicator.tsx [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+  const dropdownRef = useRef(null);
+  const { error, account } = useWeb3React();
+  const theme = useTheme();
+
+  const [showDropdown, setShowDropdown] = React.useState(false);
+  const [dropdownValues, setDropdownValues] = React.useState([]);
+  const [currentChainId, setCurrentChainId] = React.useState('');
+
+  // fetches the current chain Id and sets the current chain Id state
+  const fetchCurrentChainId = async () => {
```
[the remark this answers is not in the document]
Comment (madhurg-epns) [reply src/components/ChainIndicator.tsx:31]: @mishramonalisha76 you want me to add types to the useState call right?

**Review by mishramonalisha76 — DISMISSED** [review]
LGTM

**Review by 0xNilesh — APPROVED** [review]

File: src/components/ChainIndicator.tsx [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+    });
+    setCurrentChainId(currentChainId);
+  };
+
+  // handles network change request
+  const handleChangeNetwork: (chainId: number) => Promise<void> = async (chainId) => {
+    const chainIds = appConfig?.allowedNetworks;
+    if (chainIds.includes(chainId)) {
+      try {
+        await window.ethereum.request({
```
Comment (0xNilesh) [inline src/components/ChainIndicator.tsx:43]: Use library or provider instead of window.ethereum.

**Review by 0xNilesh — CHANGES_REQUESTED** [review]

**Review by 0xNilesh — APPROVED** [review]

**Pull request merged**